system/inc/system_cloud.h

```diff
@@ -85,17 +85,17 @@ String spark_deviceID(void);
 extern "C" {
 #endif
 
-enum cloud_disconnect_reason {
-    CLOUD_DISCONNECT_REASON_OTHER = 0,
+typedef enum cloud_disconnect_reason {
```

Comment:
Should we have a cloud disconnect reason `CLOUD_DISCONNECT_NETWORK_DOWN` or similar?